addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/listeners/RetentionBusinessEventListener.java

```diff
@@ -78,4 +84,20 @@ public void handleEvent(Event event) {
         }
     }
 
+    protected boolean isExpressionLanguage(String expression) {
+        RepositoryService repositoryService = Framework.getService(RepositoryService.class);
+        UserManager userManager = Framework.getService(UserManager.class);
+
+        try (CloseableCoreSession session = CoreInstance.openCoreSession(repositoryService.getRepositoryNames().get(0),
```

Comment:
`openCoreSessionSystem(null)` is enough. Although using the default repository (`null`) is a sign that something is wrong, what happens if there are several repositories? Something should be passed as argument or found in the context to decide which one to use.

Reply:
thx, I have a few questions:
- `openCoreSessionSystem(null)` will use the defaut repo. In the case of multi repo, we always have this notion of default ? 

> is a sign that something is wrong, what happens if there are several repositories? Something should be passed as argument or found in the context to decide which one to use.

You mean in the EL context ? If I understand correctly, perhaps I am wrong, but the EL will evaluate the document as a regular pojo and will calls the method on the object, Is it rigt ? if it is the case. Where the session will be used ?

Reply:
There's always a default repo, this is configurable in repo contribs, and defaults to the "default" repo.
By context I mean whatever context this is being run in, for example operation context, or object context. If you have a session field then use its repository.

Reply:
Given a re-reading of the code, the `isExpression` check depends on the repository, so the query construction must be moved inside the loop over all repositories.